datafusion/physical-expr/src/utils.rs

```diff
@@ -196,27 +199,156 @@ pub fn normalize_sort_requirement_with_equivalence_properties(
     }
 }
 
+pub fn normalize_expr_with_ordering_equivalence_properties(
```

Comment:
I wonder if there is some way to reduce the code replication here -- SortRequirement and SortExpr are very similar, as are `OrderingEquivalentClass` and `EquivalentClass`